tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts

```diff
@@ -179,6 +181,7 @@ export class ScalarCardContainer implements CardRenderer, OnInit, OnDestroy {
   chartMetadataMap$?: Observable<ScalarCardSeriesMetadataMap>;
   selectedTime$?: Observable<ViewSelectedTime | null>;
   internalSelectedTime$?: Observable<LinkedTime | null>;
+  minMaxSteps$?: Observable<MinMaxStep | null>;
```

Comment:
We don't have/use nonNullAsync pipe in the open source code base. This is mainly because the compiler we use in the open source repo doesn't enforce the possible nullness of the regular async pipe. 

Instead, in this code base, we just instead rely on runtime errors to implicitly tell us when the async pipe is unexpectedly returning null values and then decide what to do about it.

I can't reason out if this will actually be non-Null in practice. If neither of you can also reason out whether it should be null then I'd say remove the null from the type and remove any code that handles a null value and see if you get any runtime errors.

Reply:
My thought was that this will be null when dataSeries$ is null since it relies on the value sent out from that Observable. I don't know when that would be null but, that is why I left the null in there.

Reply:
Have you observed dataSeries$ being null? It's typed as Observable<ScalarCardDataSeries[]> so the compiler, at least, doesn't think it can be null.

I'd recommend we remove the null unless we observe that it can actually be null. In #5801 I am able to remove the null portion of the type -- compiler doesn't complain and there don't appear to be any runtime errors.